apps/web/app/.server/service/proposals.ts

```diff
@@ -48,10 +48,21 @@ export async function getProposalsFromL1() {
   // Logs are calculated from the last 40 days,
   // as this is a conservative estimation of oldest block with a valid upgrade.
   const blocksInADay = Math.floor((24 * 60 * 60) / EthereumConfig.l1.blockTime);
-  const from = bigIntMax(currentBlock - BigInt(40 * blocksInADay), BigInt(0));
+  const minimumPossibleBlock = bigIntMax(currentBlock - BigInt(40 * blocksInADay), BigInt(0));
+  const newestKnownBlock =
+    storedProposals
+      .map((p) => BigInt(p.proposedOn.valueOf()) / 1000n)
+      .sort()
+      .map((timestamp) => {
+        return (
+          currentBlock - (latestBlock.timestamp - timestamp) / BigInt(EthereumConfig.l1.blockTime)
+        );
+      })
+      .at(-1) || minimumPossibleBlock;
+
   const logs = await getUpgradeStartedEvents({
-    fromBlock: from,
-    toBlock: "latest",
+    fromBlock: bigIntMax(minimumPossibleBlock, newestKnownBlock),
```

Comment:
It is fine, but maybe choosing min is better here? In case some proposal was accidentally missed, it could be refetched here

Reply:
The logic here is this one:
- We know that upgrades live at top 40 days.
- But if we know in our db an upgrade newer than that, that means that we instead of fetching 40 days in the past, we can fetch from the moment that upgrade was published.
- That's why we get the max, because we want to newer of the 2.